File: src/main/java/com/dh/ondot/member/application/command/CreateChoicesCommand.java
package com.dh.ondot.member.application.command;

import com.dh.ondot.member.api.request.OnboardingRequest;

import java.util.List;

public record CreateChoicesCommand(
        List<QuestionAnswerPair> questionAnswerPairs
) {
    public record QuestionAnswerPair(
            Long questionId,
            Long answerId
    ) {
    }
    
    public static CreateChoicesCommand from(OnboardingRequest request) {
        List<QuestionAnswerPair> pairs = request.questions().stream()
                .map(q -> new QuestionAnswerPair(q.questionId(), q.answerId()))
                .toList();
        return new CreateChoicesCommand(pairs);
    }
}

File: src/main/kotlin/com/dh/ondot/member/application/command/CreateChoicesCommand.kt
package com.dh.ondot.member.application.command

import com.dh.ondot.member.presentation.request.OnboardingRequest

data class CreateChoicesCommand(
    val questionAnswerPairs: List<QuestionAnswerPair>,
) {
    data class QuestionAnswerPair(
        val questionId: Long,
        val answerId: Long,
    )

    companion object {
        fun from(request: OnboardingRequest): CreateChoicesCommand {
            val pairs = request.questions.map { q ->
                QuestionAnswerPair(q.questionId, q.answerId)
            }
            return CreateChoicesCommand(pairs)
        }
    }
}
